frontend/src/components/ui/form.tsx

```diff
@@ -0,0 +1,177 @@
+import * as React from 'react';
+import * as LabelPrimitive from '@radix-ui/react-label';
+import { Slot } from '@radix-ui/react-slot';
+import {
+  Controller,
+  ControllerProps,
+  FieldPath,
+  FieldValues,
+  FormProvider,
+  useFormContext,
+} from 'react-hook-form';
+
+import { cn } from '@/lib/utils';
+import { Label } from '@/components/ui/label';
+
+const Form = FormProvider;
+
+type FormFieldContextValue<
+  TFieldValues extends FieldValues = FieldValues,
+  TName extends FieldPath<TFieldValues> = FieldPath<TFieldValues>,
+> = {
+  name: TName;
+};
+
+const FormFieldContext = React.createContext<FormFieldContextValue>(
+  {} as FormFieldContextValue
+);
+
+const FormField = <
+  TFieldValues extends FieldValues = FieldValues,
+  TName extends FieldPath<TFieldValues> = FieldPath<TFieldValues>,
+>({
+  ...props
+}: ControllerProps<TFieldValues, TName>) => {
+  return (
+    <FormFieldContext.Provider value={{ name: props.name }}>
+      <Controller {...props} />
+    </FormFieldContext.Provider>
+  );
+};
+
+const useFormField = () => {
+  const fieldContext = React.useContext(FormFieldContext);
+  const itemContext = React.useContext(FormItemContext);
+  const { getFieldState, formState } = useFormContext();
+
+  const fieldState = getFieldState(fieldContext.name, formState);
+
+  if (!fieldContext) {
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/19#issuecomment-2379954551>#1588a1</a><span> - 09/27/2024, 08:00 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The useFormField hook throws an error if used outside of a FormField context, which might be too restrictive and could lead to unexpected crashes.
</div>
<div id="fix">
<b>Fix:</b> Instead of throwing an error, consider returning a default or null value when the context is not available, and handle this case in the components that use the hook.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -46,9 +46,13 @@
    const fieldState = getFieldState(fieldContext.name, formState);

    if (!fieldContext) {
 -    throw new Error('useFormField should be used within <FormField>');
 +    console.warn('useFormField is being used outside of <FormField>');
 +    return {
 +      name: '',
 +      id: '',
 +      formItemId: '',
 +      formDescriptionId: '',
 +      formMessageId: '',
 +    };
    }

    const { id } = itemContext;
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged

Reply:
this is shadcn code, will leave it as it is